qa/tasks/deepsea.py

```diff
@@ -363,8 +404,64 @@ def _introspect_roles(self, deepsea_ctx):
         for var in report_vars:
             self.log.info("{} == {}".format(var, deepsea_ctx[var]))
 
+    def _master_python_version(self, py_version):
+        """
+        Determine if a given python version is installed on the Salt Master
+        node.
+        """
+        python_binary = 'python{}'.format(py_version)
+        installed = True
+        try:
+            self.master_remote.run(args=[
+                'type',
+                python_binary,
+                run.Raw('>'),
+                '/dev/null',
+                run.Raw('2>&1'),
+                ])
+        except CommandFailedError:
+            installed = False
+        if installed:
```

Comment:
No sense in this 'if', because you can run python binary with --version option in try-catch block earlier instead of invoking 'type', since it is equivalent operations.

Reply:
The sole purpose of the existing try block is to determine if the python binary is installed. If I add additional commands to it, I can't be sure which command failed and for what reason.